src/Microsoft.Azure.Functions.Extensions.Mcp/Trigger/McpToolListener.cs

```diff
@@ -64,7 +69,47 @@ public async Task<CallToolResult> RunAsync(CallToolRequestParams callToolRequest
                 }
             ]
         };
+    }
 
+    internal static void ValidateArgumentsHaveRequiredProperties(ICollection<IMcpToolProperty> properties, CallToolRequestParams callToolRequest)
+    {
+        var requiredProps = properties?
+                            .Where(p => p.Required)
+                            .ToArray();
 
+        if (requiredProps is null || requiredProps.Length == 0)
+        {
+            return;
+        }
+
+        var missing = new List<string>();
+        var args = callToolRequest?.Arguments;
+
+        foreach (var prop in requiredProps)
```

Comment:
Just spoke to Fabio about this and he had a good point: the shortcircut on lines 80 doesn't help us as the array assignment on line 76 already iterates through the entire properties list to get check of `Required`. So with the foreach here we end up iterating twice.

So the suggestion here is to remove lines 76-78; revert back to checking properties null/length on line 80, and update this foreach loop to check for prop.Required and continue if false